frontend/src/metabase/admin/settings/components/widgets/AdminSettingInput.unit.spec.tsx

```diff
@@ -0,0 +1,420 @@
+import { userEvent } from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+
+import {
+  setupPropertiesEndpoints,
+  setupSettingsEndpoints,
+  setupUpdateSettingEndpoint,
+} from "__support__/server-mocks";
+import {
+  fireEvent,
+  mockScrollIntoView,
+  renderWithProviders,
+  screen,
+  waitFor,
+} from "__support__/ui";
+import { UndoListing } from "metabase/containers/UndoListing";
+import type { SettingKey } from "metabase-types/api";
+import {
+  createMockSettingDefinition,
+  createMockSettings,
+} from "metabase-types/api/mocks";
+
+import {
+  AdminSettingInput,
+  type AdminSettingInputProps,
+} from "./AdminSettingInput";
+
+const setup = (props: AdminSettingInputProps<SettingKey>) => {
+  setupPropertiesEndpoints(
+    createMockSettings({
+      "site-name": "Metabased",
+      "site-url": "http://localhost:8675309",
+      "enable-xrays": true,
+      "humanization-strategy": "none",
+      "query-caching-min-ttl": 64,
+      "premium-embedding-token": "super-secret",
+      // @ts-expect-error - this isn't a valid setting
+      "fake-setting": "fake-value",
+    }),
+  );
+
+  setupSettingsEndpoints([
+    createMockSettingDefinition({
+      key: "site-url",
+      is_env_setting: true,
+      env_name: "MB_HARDCODED_URL",
+    }),
+  ]);
+
+  setupUpdateSettingEndpoint();
+  renderWithProviders(
+    <>
+      <AdminSettingInput {...props} />
+      <UndoListing />
+    </>,
+  );
+};
+
+describe("AdminSettingInput", () => {
+  beforeAll(() => {
+    mockScrollIntoView();
+  });
+
+  it("should not allow invalid settings", async () => {
+    setup({
+      title: "Fake Setting",
+      // @ts-expect-error - this isn't a valid setting
+      name: "fake-name",
+    });
+
+    // this is just a test of the types, it shouldn't fail at run time
+    expect(await screen.findByText("Fake Setting")).toBeInTheDocument();
```

Comment:
Maybe I don't understand this test properly, but why do we need to test the types?

Reply:
the real assertion is in the `@ts-expect-error` - if someone loosens up the type to allow invalid settings it'll throw a type error